### DISCUSSION TITLE
What is the docker image that is being used when we create an instance using SkyPilot

### DISCUSSION THREAD

**Discussion by hessiowork** [skypilot-org/skypilot#3812] [Ideas]
I was wondering what container image is being used on the instance that is rented when you run a job using SkyPilot? Can/How would you change the image used?

**Comment by chris-aeviator** [comment]
trying to find this info myself (seems undocumented :/ ) - the skypilot team publishes https://hub.docker.com/r/berkeleyskypilot/skypilot .

When I do `sky launch --cloud kubernetes` I can see my kubernetes pulling ` us-central1-docker.pkg.dev/skypilot-375900/skypilotk8s/skypilot:20240613`

**Reply by romilbhardwaj** [reply]
Thanks for the report, we are adding this to our docs: https://github.com/skypilot-org/skypilot/pull/4129/files

**Comment by romilbhardwaj** [comment]
On clouds, we do not run in a container by default. The base VM image depends on the clouds chosen - you can refer to images.csv in our catalog: https://github.com/skypilot-org/skypilot-catalog/tree/master/catalogs/v5

On Kubernetes, we maintain and use two SkyPilot container images:

1. `us-central1-docker.pkg.dev/skypilot-375900/skypilotk8s/skypilot`: used for CPU-only clusters ([Dockerfile](https://github.com/skypilot-org/skypilot/blob/master/Dockerfile_k8s)).
2. ``us-central1-docker.pkg.dev/skypilot-375900/skypilotk8s/skypilot-gpu``: used for GPU clusters ([Dockerfile](https://github.com/skypilot-org/skypilot/blob/master/Dockerfile_k8s_gpus)).

These images are pre-installed with SkyPilot dependencies for fast startup.
